### DISCUSSION TITLE
Does jetstream work with the embedded test server?

### DISCUSSION THREAD

**Discussion by jkwatson-verta** [nats-io/nats-server#3164] [Q&A] [answered]
I'm trying to use the test server from `"github.com/nats-io/nats-server/test"`. It works fine with bare NATS, but when I try to add a jetstream stream, it fails with a "context deadline exceeded" error. There doesn't appear to be an option to enable Jetstream in the options that are available.

Sample code: 
```go
import (
	"github.com/nats-io/gnatsd/server"
	natsServer "github.com/nats-io/nats-server/test"
	"github.com/nats-io/nats.go"
	"testing"
)

func TestEmbeddedJetstream(t *testing.T) {
        opts := &natsServer.DefaultTestOptions
	opts.Port = server.RANDOM_PORT

	embeddedServer := natsServer.RunServer(opts)
	defer embeddedServer.Shutdown()

	serverAddress := embeddedServer.Addr().String()
	connection, err := nats.Connect("nats://"+serverAddress, nats.RetryOnFailedConnect(true))
	if err != nil {
		t.Fatal(err)
	}

	jetstream, err := connection.JetStream()
	if err != nil {
		t.Fatal(err)
	}
	_, err = jetstream.AddStream(&nats.StreamConfig{Name: "test_stream"})
	if err != nil {
		t.Fatal(err)
	}
}
```

**Comment by derekcollison** [comment, marked as answer]
Change `"github.com/nats-io/gnatsd/server"` to `"github.com/nats-io/nats-server/v2/server"`

**Reply by jkwatson-verta** [reply]
That doesn't work, because `test.RunServer()` returns the gnatsd server. Are you saying not to use `github.com/nats-io/nats-server/test` at all?

**Reply by derekcollison** [reply]
```go
import natsServer "github.com/nats-io/nats-server/v2/test"
```

Try that..

**Reply by jkwatson-verta** [reply]
awesome. that worked. Thanks a million!

**Comment by a-h** [comment]
I couldn't quite work out from the sample how to get this working, but I found https://github.com/nats-io/nats.go/issues/467#issuecomment-1771424369 and got that up and running.
